### ISSUE TITLE
problem installing on Mac with Mamba

### ISSUE THREAD

**Issue by tom-farr** [DOI-USGS/ISIS3#4677]
**ISIS version(s) affected**: 4 and 6 (?)

**Description**  
Sorry, but I'm inexperienced at installing software in this way. I tried to follow the instructions at https://github.com/USGS-Astrogeology/ISIS3/blob/dev/README.md. They seem to be outdated, but I followed and installed Anconda3 which appeared to install python 3.6 or 3.8. Following the instructions, I did 'condo create - n isis python=3.6' activated isis, config, and installed mamba. But when I tried 'mamba install -c uses-astrogeology isis' I got an error: 

Encountered problems while solving.
Problem: package python-3.6.13-haf480d7_2_cpython requires openssl >=1.1.1l,<1.1.2a, but none of the providers can be installed

I checked and eventually found openssl 1.1.1l, but it still wouldn't solve. So I went back and ran: 

conda create -n isis python=3.8

when I then ran: 'mamba install -c usgs-astrogeology isis', it appeared to load isis 6.0.0 which complained: 

Problem: package isis-6.0.0-0 requires opencv >=3.2.0,<3.3.0a0, but none of the providers can be installed 

I know these types of software are moving targets (I worked for a long time with JPL's InSAR processors), but unfortunately, I'm not sophisticated enough to solve this problem. Is there an easy solution? or am I in over my head? 

Thanks for any help. 
Tom Farr

**Additional context**  
Mac Pro laptop running OS 11.6.

**Comment by AustinSanders** [comment]
@tom-farr You're definitely not alone in experiencing installation difficulties, and getting this fixed is among the development team's highest priorities.

[Issue 4662](https://github.com/USGS-Astrogeology/ISIS3/issues/4662) describes similar difficulties, and it includes some discussion on how the user was able to resolve the problem.   Do you mind working through those steps and reporting back?

**Comment by tom-farr** [comment]
Thanks. Will do.

————
Tom Farr
Montecito, CA

On Nov 9, 2021, at 1:38 PM, AustinSanders ***@***.***> wrote:

﻿

@tom-farr<https://github.com/tom-farr> You're definitely not alone in experiencing installation difficulties, and getting this fixed is among the development team's highest priorities.

Issue 4662<https://github.com/USGS-Astrogeology/ISIS3/issues/4662> describes similar difficulties, and it includes some discussion on how the user was able to resolve the problem. Do you mind working through those steps and reporting back?

**Comment by tom-farr** [comment]
Hi Austin. it worked! Mostly. everything loaded and I then tried to run an isis routine. I've never run isis, so this could be user error, but I get the complaint that ISISROOT isn't set. However, I ran the default python thing: 

(isis6.0.0) Toms-MacBook-Pro:~ tom_farr$ python $CONDA_PREFIX/scripts/isisVarInit.py

and anyway, do I have isis activated? And so I just type the command, like so: 

(isis6.0.0) Toms-MacBook-Pro:~ tom_farr$ std2isis from=/Volumes/01TGF/Data/Radars/Sentinel-1/SJV/Asc137/++20200414_NewARIA/Asc137_LOSin_crop.tif to=temparia.cub

Please set ISISROOT before running any Isis applications

thanks, 
Tom Farr

**Comment by jlaura** [comment]
@tom-farr When creating a new env, I have had to run the conda prefix script, deactivate the environment and then reactivate the environment. Then everything worked as expected.

As an aside - are trying to process Sentinel-1 data through ISIS? Our primary support is for planetary science and not earth science missions.

**Comment by tom-farr** [comment]
Thanks- that worked, though the last steps listed in the GitHub install page are rather confusing- especially when it wanders off into talking about Docker, etc. I pieced together that one has to deactivate twice to exit isis and then conda and then activate twice, using isis6.0.0 as the second env to activate. I now seem to be able to use isis!

As for the Sentinel data- I’d like to do some post-processing on InSAR time series I processed when I was at JPL (retired now!). The time series are multi band geotiffs where each ‘band’ is a date and the floating-point pixel values are the deformation from the first date. So I thought I’d try to load those into an isis cube. I tried again the std2isis routine, but isis doesn’t like the input format, so I’ll have to keep plugging away at trying to get isis to accept my data. If you have any ideas, I’d be happy to hear them, but I do appreciate your holding the hand of a newbie. I used vicar a lot in the old days and can see the similarity, but every software package has its quirks…

Tom Farr
_______
Tom Farr
***@***.******@***.***>




On Nov 10, 2021, at 4:48 AM, jlaura ***@***.******@***.***>> wrote:



@tom-farr<https://github.com/tom-farr> When creating a new env, I have had to run the conda prefix script, deactivate the environment and then reactivate the environment. Then everything worked as expected.

As an aside - are trying to process Sentinel-1 data through ISIS? Our primary support is for planetary science and not earth science missions.

**Comment by tom-farr** [comment]
Ok, to further demonstrate my poor mastery of unix, it appears that isis isn’t finding the ISISDATA location. I tried printenv to see what was set, but I can’t find any mention of ISISDATA. I did run the python thing in the GitHub instructions- did it not take? Or am I hopelessly lost…
Thanks ,
Tom
_______
Tom Farr
***@***.******@***.***>




On Nov 10, 2021, at 8:11 AM, Farr, Thomas ***@***.******@***.***>> wrote:

Thanks- that worked, though the last steps listed in the GitHub install page are rather confusing- especially when it wanders off into talking about Docker, etc. I pieced together that one has to deactivate twice to exit isis and then conda and then activate twice, using isis6.0.0 as the second env to activate. I now seem to be able to use isis!

As for the Sentinel data- I’d like to do some post-processing on InSAR time series I processed when I was at JPL (retired now!). The time series are multi band geotiffs where each ‘band’ is a date and the floating-point pixel values are the deformation from the first date. So I thought I’d try to load those into an isis cube. I tried again the std2isis routine, but isis doesn’t like the input format, so I’ll have to keep plugging away at trying to get isis to accept my data. If you have any ideas, I’d be happy to hear them, but I do appreciate your holding the hand of a newbie. I used vicar a lot in the old days and can see the similarity, but every software package has its quirks…

Tom Farr
_______
Tom Farr
***@***.******@***.***>




On Nov 10, 2021, at 4:48 AM, jlaura ***@***.******@***.***>> wrote:



@tom-farr<https://github.com/tom-farr> When creating a new env, I have had to run the conda prefix script, deactivate the environment and then reactivate the environment. Then everything worked as expected.

As an aside - are trying to process Sentinel-1 data through ISIS? Our primary support is for planetary science and not earth science missions.

**Comment by tom-farr** [comment]
Oops. I didn’t do printenv in the window where isis was active. Got it now.

_______
Tom Farr
***@***.******@***.***>




On Nov 13, 2021, at 11:30 AM, Farr, Thomas ***@***.******@***.***>> wrote:

Ok, to further demonstrate my poor mastery of unix, it appears that isis isn’t finding the ISISDATA location. I tried printenv to see what was set, but I can’t find any mention of ISISDATA. I did run the python thing in the GitHub instructions- did it not take? Or am I hopelessly lost…
Thanks ,
Tom
_______
Tom Farr
***@***.******@***.***>




On Nov 10, 2021, at 8:11 AM, Farr, Thomas ***@***.******@***.***>> wrote:

Thanks- that worked, though the last steps listed in the GitHub install page are rather confusing- especially when it wanders off into talking about Docker, etc. I pieced together that one has to deactivate twice to exit isis and then conda and then activate twice, using isis6.0.0 as the second env to activate. I now seem to be able to use isis!

As for the Sentinel data- I’d like to do some post-processing on InSAR time series I processed when I was at JPL (retired now!). The time series are multi band geotiffs where each ‘band’ is a date and the floating-point pixel values are the deformation from the first date. So I thought I’d try to load those into an isis cube. I tried again the std2isis routine, but isis doesn’t like the input format, so I’ll have to keep plugging away at trying to get isis to accept my data. If you have any ideas, I’d be happy to hear them, but I do appreciate your holding the hand of a newbie. I used vicar a lot in the old days and can see the similarity, but every software package has its quirks…

Tom Farr
_______
Tom Farr
***@***.******@***.***>




On Nov 10, 2021, at 4:48 AM, jlaura ***@***.******@***.***>> wrote:



@tom-farr<https://github.com/tom-farr> When creating a new env, I have had to run the conda prefix script, deactivate the environment and then reactivate the environment. Then everything worked as expected.

As an aside - are trying to process Sentinel-1 data through ISIS? Our primary support is for planetary science and not earth science missions.

**Comment by tom-farr** [comment]
Sorry to keep pestering you. I’m now trying to run some simple routines on known data. I move the data into the ISISDATA place and run std2isis. No complaints, but I can’t find the output file. qview won’t find it either. I’m sure it’s something simple…

(isis6.0.0) Toms-MacBook-Pro:~ tom_farr$ std2isis from=c1.png to=testc1.cub

_______
Tom Farr
***@***.******@***.***>




On Nov 13, 2021, at 12:10 PM, Farr, Thomas ***@***.******@***.***>> wrote:

Oops. I didn’t do printenv in the window where isis was active. Got it now.

_______
Tom Farr
***@***.******@***.***>




On Nov 13, 2021, at 11:30 AM, Farr, Thomas ***@***.******@***.***>> wrote:

Ok, to further demonstrate my poor mastery of unix, it appears that isis isn’t finding the ISISDATA location. I tried printenv to see what was set, but I can’t find any mention of ISISDATA. I did run the python thing in the GitHub instructions- did it not take? Or am I hopelessly lost…
Thanks ,
Tom
_______
Tom Farr
***@***.******@***.***>




On Nov 10, 2021, at 8:11 AM, Farr, Thomas ***@***.******@***.***>> wrote:

Thanks- that worked, though the last steps listed in the GitHub install page are rather confusing- especially when it wanders off into talking about Docker, etc. I pieced together that one has to deactivate twice to exit isis and then conda and then activate twice, using isis6.0.0 as the second env to activate. I now seem to be able to use isis!

As for the Sentinel data- I’d like to do some post-processing on InSAR time series I processed when I was at JPL (retired now!). The time series are multi band geotiffs where each ‘band’ is a date and the floating-point pixel values are the deformation from the first date. So I thought I’d try to load those into an isis cube. I tried again the std2isis routine, but isis doesn’t like the input format, so I’ll have to keep plugging away at trying to get isis to accept my data. If you have any ideas, I’d be happy to hear them, but I do appreciate your holding the hand of a newbie. I used vicar a lot in the old days and can see the similarity, but every software package has its quirks…

Tom Farr
_______
Tom Farr
***@***.******@***.***>




On Nov 10, 2021, at 4:48 AM, jlaura ***@***.******@***.***>> wrote:



@tom-farr<https://github.com/tom-farr> When creating a new env, I have had to run the conda prefix script, deactivate the environment and then reactivate the environment. Then everything worked as expected.

As an aside - are trying to process Sentinel-1 data through ISIS? Our primary support is for planetary science and not earth science missions.

**Comment by jlaura** [comment]
@tom-farr You do not need to move data into the $ISISDATA area in order to process it. In fact, I would suggest not moving your data into $ISISDATA. The $ISISDATA directory is for data that ISIS needs in order to run.

Based on your command, it looks like you are running `std2isis` from your home directory. I would expect the output to be in the same directory since the `to=` command does not specify any path and it relative to where you ran the command. If that does not work, since you are on a Mac, you can click the magnifying glass on the upper right hand side of the menu bar and search for 'testc1.cub'.

**Comment by tom-farr** [comment]
Oof. Guess I deserved that. I do know my way around a Mac having used them since they were born, so I did the search as you suggest as soon as no file showed up in the top directory. It ain’t there. But no error message from isis either… 
Tom
————
Tom Farr
Montecito, CA

**Comment by jlaura** [comment]
@tom-farr That is quite strange as I would expect ISIS to complain loudly if something fails. What about specifying an absolute PATH in your to= argument?

**Comment by jessemapel** [comment]
It sounds like you need to download the data area, at least the base data.

https://github.com/USGS-Astrogeology/ISIS3/blob/dev/README.md#The-ISIS-Data-Area

That is what goes into $ISISDATA.

I would try re-running your isis2std command with a full path. Also look in the directory where you ran isis2std, there should be a print.prt file that contains logging information about all ISIS programs that you ran in that directory.

**Comment by tom-farr** [comment]
Ok, specifying the paths worked. I didn’t have to download any of the isis base data. It wouldn’t work if I put the input data into ISISDATA. And it wouldn’t work if I specified the input path which had a ++ starting the name of the directory where the data were. I didn’t know unix didn’t like that- I use it to force the directory name to be at the top of the alphabetized list. I used a different directory and std2isis worked. qview also displayed the image. Yay!
Thanks for all your patience! 
Tom
_________
Tom Farr
Montecito, CA
